syncing/src/main/kotlin/maru/syncing/pipeline/DownloadCompleteBlockRangeTask.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.syncing.pipeline
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import java.util.function.Function
+import kotlin.time.Duration
+import maru.core.SealedBeaconBlock
+import maru.p2p.MaruPeer
+import maru.p2p.PeerLookup
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import tech.pegasys.teku.networking.p2p.peer.DisconnectReason
+import tech.pegasys.teku.networking.p2p.reputation.ReputationAdjustment
+
+class DownloadCompleteBlockRangeTask(
+  private val peerLookup: PeerLookup,
+  private val maxRetries: UInt,
+  private val blockRangeRequestTimeout: Duration,
+) : Function<SyncTargetRange, CompletableFuture<List<SealedBlockWithPeer>>> {
+  private val log: Logger = LogManager.getLogger(this::javaClass)
+
+  override fun apply(targetRange: SyncTargetRange): CompletableFuture<List<SealedBlockWithPeer>> =
+    CompletableFuture<List<SealedBeaconBlock>>.supplyAsync {
```

Comment:
```suggestion
    CompletableFuture.supplyAsync {
```

I don't think type specification is any useful here